image_transport/src/camera_publisher.cpp

```diff
@@ -68,32 +77,61 @@ struct CameraPublisher::Impl
     }
   }
 
+  std::shared_ptr<NodeType> node_;
```

Comment:
by holding a reference to the Node here, are we going to end up with some sort of hanging reference where the node never goes out of scope?\

Reply:
I'll change them into weak pointers, does that sound good to you?